core/src/sumeragi/network_topology.rs

```diff
@@ -355,188 +197,3 @@ pub enum Role {
     /// Proxy Tail.
     ProxyTail,
 }
-
-impl Role {
-    /// Returns peers that have this `Role` in this voting round.
-    pub fn peers(self, network_topology: &Topology) -> Vec<PeerId> {
-        match self {
-            Role::Leader => vec![network_topology.leader().clone()],
-            Role::ValidatingPeer => network_topology.validating_peers().to_vec(),
-            Role::ObservingPeer => network_topology.peers_set_b().to_vec(),
-            Role::ProxyTail => vec![network_topology.proxy_tail().clone()],
-        }
-    }
-}
-
-#[cfg(test)]
```

Comment:
They don't test things that are present in the new code. The new code is so simple that have dedicated tests is a very marginal benefit. You would just end up implementing it twice so that one copy of the code could test the other.